XiEditor/AppDelegate.swift

```diff
@@ -90,6 +94,20 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         return nil
     }
 
+    /// Passed an NSFontManager instance (as on a user-initiated font change)
+    /// computes the next set of drawing metrics and updates cached styles.
+    func handleFontChange(fontManager: NSFontManager) {
+        let newFont = fontManager.convert(textMetrics.font)
+        textMetrics = TextDrawingMetrics(font: newFont)
+        styleMap.updateFont(to: newFont)
+
+        for doc in NSApplication.shared().orderedDocuments {
+            guard let doc = doc as? Document else { continue }
+            doc.editViewController?.updateGutterWidth()
+            doc.editViewController?.editView.needsDisplay = true
```

Comment:
yea so if you change from a font that fits the document in the current window to a font which needs scroll, scrolling isn't available until you next send an update (like by moving the cursor)